mlflow/models/evaluation/default_evaluator.py

```diff
@@ -627,6 +632,11 @@ def _log_model_explainability(self):
             # Shap evaluation might fail on some edge cases, e.g., unsupported input data values
             # or unsupported model on specific shap explainer. Catch exception to prevent it
             # breaking the whole `evaluate` function.
+
+            # NaN errors should break evaluation
+            if str(e).find("NaN") != -1:
+                raise e
```

Comment:
Why should NaN errors break evaluation?

Reply:
I was not sure how else to catch the NaNs created by the old truncation logic. The NaNs don't carry through to the results passed via the `evaluate` function. If the user can override the exception handler to raise exceptions then this would be easier to test against.

Reply:
Does the NaN error occur in the updated code?

Reply:
No, but there should probably be a way to catch NaN errors in the tests

Reply:
How about this instead? 

```
except Exception as e:
    # Shap evaluation might fail on some edge cases, e.g., unsupported input data values
    # or unsupported model on specific shap explainer. Catch exception to prevent it
    # breaking the whole `evaluate` function.

    if not self.evaluator_config.get("ignore_exceptions"):
        raise e
```

and then the test can just include `evaluator_config={"ignore_exceptions": False}` in the `evaluate` call.